### PR TITLE
NMA-5455: Update design of upcoming workouts

### PR DESCRIPTION
**Pull request by gurooerbech**
### What's new?
* Made it so that upcoming workouts and the schedule used in Discovery use the same content composables for time/duration and workout info
* Tweak the UpcomingWorkoutListItem to now look more like the Schedule (that's used in Discovery). The biggest difference between the two is that the UpcomingWorkoutListItem can contain friends that are joining and their surrounding styling.
* Add a new button style as per the design system named WaitingListSecondary which is supposed to be used in scenarios where a member is currently booked on the waiting list of a Gx.
* Add previews for disabled buttons.
* Fix a wrong dark mode color for Waitilinglist disabled so it matches the colors in [Figma](https://www.figma.com/file/WzKCwRY09zn4rzRVfY0YvdRt/sats-ds-styles?type=design&node-id=1-2&mode=design&t=027yMpBIaZSeyTCh-0) Couldn't see that there were any usages of it other than in buttons, but maybe double check this.

### CODE REVIEW

File: sats-dna/src/main/kotlin/com/sats/dna/components/upcomingworkouts/Schedule.kt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.sats.dna.components.schedule
+package com.sats.dna.components.upcomingworkouts
```
Comment (gurooerbech) [inline sats-dna/src/main/kotlin/com/sats/dna/components/upcomingworkouts/Schedule.kt:1]: Idk if maybe Schedule deserves a bit of a name change? It is very closely tied to upcoming workouts 🤔

File: sats-dna/src/main/kotlin/com/sats/dna/components/button/SatsButtonColor.kt [hunk 2]
```diff
@@ -22,13 +22,26 @@ internal fun SatsButtonColor.asMaterialButtonColors(isEnabled: Boolean): ButtonC
...
+    @Composable get() = when (this) {
+        SatsButtonColor.WaitingListSecondary -> SatsTheme.colors.waitingList.primary
+        else -> null
+    }
+
 internal val SatsButtonColor.backgroundColor: Color
     @Composable get() = when (this) {
         SatsButtonColor.Primary -> SatsTheme.colors.primary.default
         SatsButtonColor.Cta -> SatsTheme.colors.cta.default
         SatsButtonColor.Secondary -> SatsTheme.colors.secondary.default
```
Comment (gurooerbech) [inline sats-dna/src/main/kotlin/com/sats/dna/components/button/SatsButtonColor.kt:41]: There will be updates coming to the secondary button as well. Just need to sort out whether the colors in Figma are correct. And then I'll be adding that change too.

**Review by cecilieopperudsats — APPROVED** [review]
I like it, Picasso 🎨

**Review by sindrenm — APPROVED** [review]
Very nice!

**Pull request merged**